cmd/esc/cli/env_set.go

```diff
@@ -57,17 +59,25 @@ func newEnvSetCmd(env *envCommand) *cobra.Command {
 				return fmt.Errorf("path must contain at least one element")
 			}
 
+			input := args[1]
 			var yamlValue yaml.Node
-			if err := yaml.Unmarshal([]byte(args[1]), &yamlValue); err != nil {
-				return fmt.Errorf("invalid value: %w", err)
-			}
-			if len(yamlValue.Content) == 0 {
-				// This can happen when the value is empty (e.g. when "" is present on the command line). Treat this
-				// as the empty string.
-				err = yaml.Unmarshal([]byte(`""`), &yamlValue)
-				contract.IgnoreError(err)
+			if rawString {
+				yamlValue.SetString(input)
+				if mustEscape := strings.ContainsFunc(input, func(r rune) bool { return !strconv.IsPrint(r) }); mustEscape {
+					yamlValue.Style = yaml.DoubleQuotedStyle
+				}
```

Comment:
Out of paranoia--have we verified that this works for non-printable characters? I _think_ it will--it should--but maybe we need to manually escape these characters as well as setting the literal style?

Reply:
@pgavlin Yah, this seems to work.
For example:
`go run ./cmd/esc env set delme/abc/audit a "$(echo "Hello\x05World")" --string`
becomes
`a: "Hello\x05World"` (this is correct, right?)

Reply:
Rad. Yeah that's good. Could we add that as a test case?